### PR TITLE
Add white and blacklists to filters

### PR DESCRIPTION
**Pull request by blogh**
This is an attempt to archive #15 

Negative points : (a lot) more parameters.
Positive points : the implementation is quite simple an the result is flexible.

### CODE REVIEW

File: pg_log_userqueries.c [hunk 1]
```diff
@@ -759,10 +931,15 @@ pgluq_log(const char *query)
 	else
 		logged_in_utility_hook = false;
 
-	/* when log regexp statement is set do not log the query if it doesn't match the regexp */
-	if ((log_query != NULL) && (regexec(&query_regexv, query, 0, 0, 0) != 0))
-		return;
+	/* Filter querries according to the white and balck list*/
+	if (log_query != NULL || log_query_blacklist != NULL) {
+		if ( ! pgluq_checkitem(	appname,
```
Comment (gleu) [inline pg_log_userqueries.c:936]: This doesn't compile. I get a:

```
pg_log_userqueries.c:936:27: error: ‘appname’ undeclared (first use in this function)
  936 |   if ( ! pgluq_checkitem( appname,
      |                           ^~~~~~~
```

Changing it to application_name allows the compilation to finish, but I'm not sure this is a real fix.

**Comment by gleu** [comment]
I'm OK to adding this PR (once fixed) on master. But there is the issue above, and there's a bigger one. With a very simple configuration, it simply doesn't work. Here is the config:

```
shared_preload_libraries = 'pg_log_userqueries'
pg_log_userqueries.log_level = LOG
```

With this configuration, pg_log_userqueries should log every query executed by a superuser. With this patch, it simply doesn't.

Moreover, there are a few warnings that ought to be fixed.

**Comment by blogh** [comment]
forced pushed to rebase. I suppose there is a better way to do this ...

File: pg_log_userqueries.c (hunk 1 above)
Comment (blogh) [reply pg_log_userqueries.c]: Fixed

**Comment by gleu** [comment]
Meaning that I can work on your PR?

**Comment by blogh** [comment]
Yes, I also fixed the unused variables warning.

**Comment by gleu** [comment]
Merged, thanks.

**Pull request closed without merging**